### PR TITLE
Do not reveal opaques in the param-env, we got lazy norm instead

### PR DESCRIPTION
**Pull request by compiler-errors**
r? lcnr

### CODE REVIEW

File: compiler/rustc_middle/src/ty/util.rs [hunk 1]
```diff
@@ -1751,6 +1751,11 @@ pub fn reveal_opaque_types_in_bounds<'tcx>(
     tcx: TyCtxt<'tcx>,
     val: ty::Clauses<'tcx>,
 ) -> ty::Clauses<'tcx> {
+    // No need to reveal opaques with the new solver enabled.
+    if tcx.next_trait_solver_globally() {
+        return val;
+    }
```
Comment (lcnr) [inline compiler/rustc_middle/src/ty/util.rs:1757]: vibeck: why here and not in `ParamEnv::with_reveal_all_normalized`?

Comment (compiler-errors) [reply compiler/rustc_middle/src/ty/util.rs:1757]: Ya i can do it there and then just assert here

Comment (lcnr) [reply compiler/rustc_middle/src/ty/util.rs:1757]: r=me after that

File: compiler/rustc_middle/src/ty/mod.rs [hunk 2]
```diff
@@ -1076,11 +1076,6 @@ impl<'tcx> ParamEnv<'tcx> {
         ty::ParamEnv { packed: CopyTaggedPtr::new(caller_bounds, ParamTag { reveal }) }
     }
 
-    pub fn with_user_facing(mut self) -> Self {
```
Comment (compiler-errors) [inline compiler/rustc_middle/src/ty/mod.rs:1079]: I also removed this, since it's both (1.) almost certainly wrong to ever use, and (2.) unused.

**Pull request merged**